SignalServiceKit/src/Messages/Interactions/TSIncomingMessage.m

```diff
@@ -141,10 +140,7 @@ - (void)markAsReadWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
     if (sendReadReceipt) {
         // Notification must happen outside of the transaction, else we'll likely crash when the notification receiver
```

Comment:
nit: This comment is now incorrect. 

Previously it was easier to see that nothing fancy was happening at this point within this method, which is nice because we know there is an open transaction (per the transaction param). 

Is there anything you can think of to help ensure future us do something dumb like open a new db transaction within the `messageWasReadLocally` code path? Maybe the answer is just "be careful" 😬 

FWIW I prefer the old async notification mechanism for this reason, but it's probably not going to be as performant as what you've proposed here.

This isn't a blocking concern - just an ongoing quest to get better at reasoning about our db transaction adjacent code.

Reply:
Yup, I have the same questions.  I can tell you that I've become much more careful when touching code paths that are hard to exercise in a test plan.  A code path like this that is touched by every read message I'm less worried about.

Reply:
fair point!